scipy/linalg/_backend.py

```diff
@@ -0,0 +1,98 @@
+import scipy._lib.uarray as ua
+from scipy.linalg import _api
+from scipy.sparse import issparse
+import numpy as np
+
+
+__all__ = [
+    'register_backend', 'set_backend',
+    'set_global_backend', 'skip_backend'
+]
+
+
+class scalar_tuple_callable_array:
+    """
+    Special case argument that can be either a scalar, tuple or array
+    for __ua_convert__.
+    """
+    pass
+
+class _ScipyLinalgBackend:
+    __ua_domain__ = "numpy.scipy.linalg"
+
+
+    @staticmethod
+    def __ua_function__(method, args, kwargs):
+        fn = getattr(_api, method.__name__, None)
+
+        if fn is None:
+            return NotImplemented
+
+        return fn(*args, **kwargs)
+
+
+    @ua.wrap_single_convertor
+    def __ua_convert__(value, dispatch_type, coerce):
+        if value is None or issparse(value):
+            return value
```

Comment:
Just to keep a track, this:

* overrides functions which use `asanyarray` internally, and
* breaks for sparse matrix inputs

The `asanyarray` should be looked into per function (and possibly removed, as I think is likely a good idea for clarkson_woodruff), the sparse matrices need handling with an explicit check here and don't need to be coerced.

Original comment by @rgommers : https://github.com/rgommers/scipy/pull/114#discussion_r786671330